kong/runloop/balancer.lua

```diff
@@ -834,21 +836,39 @@ do
     return false
   end
 
+  is_leader_state_updated = function()
+    local leader_state_version = kong.core_cache:get("leader-state:version", nil, noop)
+    if leader_state_version ~= nil then
+      return true
+    end
+
+    return false
+  end
+
   set_worker_state_stale = function()
     log(DEBUG, "invalidating proxy state")
     kong.core_cache:invalidate(worker_state_VERSION)
+    kong.core_cache:invalidate("leader-state:version")
   end
 
 
   set_worker_state_updated = function()
     worker_state_version = kong.core_cache:get(worker_state_VERSION, TTL_ZERO, utils.uuid)
     log(DEBUG, "proxy state is updated")
+
+    if worker_pid() == 0 then
```

Comment:
If worker 0 dies for any reason, does this number get recycled, or should a new leader get elected?